x-pack/plugins/lens/public/visualizations/metric/dimension_editor.tsx

```diff
@@ -195,8 +173,87 @@ function MaximumEditor({ setState, state }: Props) {
   );
 }
 
-function PrimaryMetricEditor(props: Props) {
-  const { state, setState, frame, accessor } = props;
+function SecondaryMetricEditor({ accessor, idPrefix, frame, layerId, setState, state }: SubProps) {
+  const columnName = getColumnByAccessor(accessor, frame.activeData?.[layerId].columns)?.name;
+  const defaultPrefix = columnName || '';
+
+  return (
+    <div data-test-subj="lnsMetricDimensionEditor_secondary_metric">
+      <EuiFormRow
+        display="columnCompressed"
+        fullWidth
+        label={i18n.translate('xpack.lens.metric.prefixText.label', {
+          defaultMessage: 'Prefix',
+        })}
+      >
+        <EuiButtonGroup
+          isFullWidth
+          buttonSize="compressed"
+          legend={i18n.translate('xpack.lens.metric.prefix.label', {
+            defaultMessage: 'Prefix',
+          })}
+          data-test-subj="lnsMetric_prefix_buttons"
+          options={[
+            {
+              id: `${idPrefix}auto`,
+              label: i18n.translate('xpack.lens.metric.prefix.auto', {
+                defaultMessage: 'Auto',
+              }),
+              'data-test-subj': 'lnsMetric_prefix_auto',
+              value: undefined,
+            },
+            {
+              id: `${idPrefix}custom`,
+              label: i18n.translate('xpack.lens.metric.prefix.custom', {
+                defaultMessage: 'Custom',
+              }),
+              'data-test-subj': 'lnsMetric_prefix_custom',
+              value: defaultPrefix,
+            },
+            {
+              id: `${idPrefix}none`,
+              label: i18n.translate('xpack.lens.metric.prefix.none', {
+                defaultMessage: 'None',
+              }),
+              'data-test-subj': 'lnsMetric_prefix_none',
+              value: '',
+            },
+          ]}
+          idSelected={`${idPrefix}${
+            state.secondaryPrefix === undefined
+              ? 'auto'
+              : state.secondaryPrefix === ''
+              ? 'none'
+              : 'custom'
+          }`}
+          onChange={(_id, secondaryPrefix) => {
+            setState({
+              ...state,
+              secondaryPrefix,
+            });
+          }}
+        />
+      </EuiFormRow>
+      {state.secondaryPrefix && (
+        <EuiFormRow label=" " fullWidth display="columnCompressed">
+          <DebouncedInput
+            compressed
+            value={state.secondaryPrefix}
+            onChange={(newPrefix) => {
+              setState({
+                ...state,
+                secondaryPrefix: newPrefix,
+              });
+            }}
+          />
+        </EuiFormRow>
```

Comment:
Rather than using a second `EuiFormRow` component with an empty `label` prop here to house the conditional text input, I think it would be better to move the input into the preceding `EuiFormRow` that contains the `EuiButtonGroup` (creating a single `EuiFormRow` that houses both the `EuiButtonGroup` and the text input). An `aria-label` should also be applied to the text input, to improve accessibility.

An example of this approach to structuring forms can be found in the EUI docs (see the "Label" example): https://elastic.github.io/eui/#/forms/compressed-forms#complex-example

Reply:
Thanks for that example. Exactly what I was looking for and couldn't find!